src/main/kotlin/dev/stalla/model/podcastindex/GeographicLocation.kt

```diff
@@ -0,0 +1,247 @@
+    public val latitude: Double,
+    public val longitude: Double,
+    public val altitude: Double? = null,
+    public val crs: String? = null,
+    public val uncertainty: Double? = null,
+    public val parameters: List<Parameter> = emptyList()
+) {
+
+    /**
+     * Instantiates a new geographic location.
+     *
+     * @property latitude The latitude value.
+     * @property longitude The longitude value.
+     * @property altitude The altitude value.
+     * @property crs The coordinate reference system - defaults to ([CRS_WGS84]) if not set.
+     * @property uncertainty The amount of uncertainty in the location as a value in meters.
+     * @property parameters The list of geo URI parameters as key/value pairs.
+     */
+    public constructor(
+        latitude: Double,
+        longitude: Double,
+        altitude: Double?,
+        crs: String?,
+        uncertainty: Double?,
+        parameters: Map<String, String>
+    ) : this(
+        latitude = latitude,
+        longitude = longitude,
+        altitude = altitude,
+        crs = crs,
+        uncertainty = uncertainty,
+        parameters = parameters.map { Parameter(it.key, it.value) }.asUnmodifiable()
+    )
+
+    /**
+     * Represents a single value parameter.
+     *
+     * @property key The key of parameter.
+     * @property value The value of parameter.
+     */
+    public data class Parameter(val key: String, val value: String) {
+        override fun equals(other: Any?): Boolean {
+            return other is Parameter &&
+                other.key.equals(key, ignoreCase = true) &&
+                other.value == value
+        }
+
+        override fun hashCode(): Int {
+            var result = key.toLowerCase(Locale.ROOT).hashCode()
+            result += 31 * result + value.toLowerCase(Locale.ROOT).hashCode()
+            return result
+        }
+    }
+
+    /**
+     * The first value for the parameter with [key] comparing
+     * case-insensitively or `null` if no such parameters found.
+     */
+    public fun parameter(key: String): String? =
+        parameters.firstOrNull { it.key.equals(key, ignoreCase = true) }?.value
+
+    /** Creates a copy of `this` type with an added parameter of [key] and [value]. */
+    public fun withParameter(key: String, value: String): GeographicLocation {
+        if (key.isBlank() || key.containsMediaTypeSeparatorSymbol()) return this
+        if (value.isBlank()) return this
+        if (hasParameter(key, value)) return this
+
+        return GeographicLocation(latitude, longitude, altitude, crs, uncertainty, parameters + Parameter(key, value))
+    }
+
+    /** Creates a copy of `this` type without any parameters.*/
+    public fun withoutParameters(): GeographicLocation = GeographicLocation(latitude, longitude, altitude)
+
+    /** Checks if `this` type matches a [pattern] type taking parameters into account. */
+    public fun match(pattern: GeographicLocation?): Boolean {
```

Comment:
I've used `match` here to match (pun intended) the corresponding method in `MediaType`, which we've adopted from Ktor. I agree, `matchs` makes more sense, but we'd have to deprecate it in `MediaType` as well then.

Reply:
Ah, sounds like too much faff for one letter, but on the other hand, maybe better to do the deprecation asap (in a separate PR tho, in case). What do you reckon?

Reply:
Yeah, let's do it right now, just to have it done. Since #93 is about deprecating stuff anyway, I can to it there.